packages/broker/src/plugins/storage/StorageConfig.ts

```diff
@@ -118,40 +124,62 @@ export class StorageConfig {
         }
     }
 
-    private _addStreams(streamKeys: Set<StreamKey>): void {
-        logger.info('Add %d streams to storage config: %s', streamKeys.size, Array.from(streamKeys).join(','))
-        this.streamKeys = new Set([...this.streamKeys, ...streamKeys])
+    private _addStreams(keysToAdd: Set<StreamKey>): void {
+        logger.info('Add %d streams to storage config: %s', keysToAdd.size, Array.from(keysToAdd).join(','))
+        this.streamKeys = new Set([...this.streamKeys, ...keysToAdd])
         this.listeners.forEach((listener) => {
-            streamKeys.forEach((key: StreamKey) => listener.onStreamAdded(getStreamFromKey(key)))
+            keysToAdd.forEach((key: StreamKey) => listener.onStreamAdded(getStreamFromKey(key)))
         })
     }
 
-    private _removeStreams(streamKeys: Set<StreamKey>): void {
-        logger.info('Remove %d streams from storage config: %s', streamKeys.size, Array.from(streamKeys).join(','))
-        this.streamKeys = new Set([...this.streamKeys].filter((x) => !streamKeys.has(x)))
+    private _removeStreams(keysToRemove: Set<StreamKey>): void {
+        logger.info('Remove %d streams from storage config: %s', keysToRemove.size, Array.from(keysToRemove).join(','))
+        this.streamKeys = new Set([...this.streamKeys].filter((x) => !keysToRemove.has(x)))
         this.listeners.forEach((listener) => {
-            streamKeys.forEach((key: StreamKey) => listener.onStreamRemoved(getStreamFromKey(key)))
+            keysToRemove.forEach((key: StreamKey) => listener.onStreamRemoved(getStreamFromKey(key)))
         })
     }
 
+    private belongsToMeInCluster(key: StreamKey): boolean {
+        const hashedIndex = Protocol.Utils.keyToArrayIndex(this.clusterSize, key.toString())
+        return hashedIndex === this.myIndexInCluster
```

Comment:
So if the cluster size changes, data may go to a different node, will this break data retrieval?

Reply:
It won't break data retrieval, because all the data still goes into the same Cassandra cluster. However, all nodes must be reconfigured and restarted if the cluster size is changed, so that they are aligned on who subscribes to what.

Reply:
Ahh I didn't catch the "same Cassandra cluster" part, makes sense now.